### PR TITLE
Jetpack Focus: Implement design for the new users overlay

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #19453

## Description
This PR implements the overlay for new users. The design for this overlay is different from the existing overlays, which required adding a way to customize the generic overlay.

## ⚠️ Disclaimer

The focus of this PR is only on the design of the overlay.
The menu card and showing the overlay on login will be implemented in a future PR.

### Screenshots

| Light Mode | Dark Mode |
| - | - |
|light|dark|

## Pre-testing Instructions
- To disable the learn more button:
    - Go to `JetpackFullscreenOverlayGeneralViewModel.swift` line 183
    - Replace this line with `return nil`
    - This won't be needed in production since the learn more button will be disabled remotely by then
- To always show the overlay even if it was shown before:
    - Go to `JetpackFeaturesRemovalCoordinator.swift` line 132
    - Change the default value to `true`

## Testing Instructions

1. Open the app
2. Open the debug menu and enable "Jetpack Features Removal Phase for New Users"
3. Navigate to My Site
4. Close and re-open the app
5. The overlay should be displayed
6. Make sure that the overlay matches the design

## Regression Notes
1. Potential unintended areas of impact
The design of other overlays.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel.swift [hunk 1]
```diff
@@ -287,6 +312,9 @@ private extension JetpackFullscreenOverlayGeneralViewModel {
             static let notificationsContinueButtonTitle = NSLocalizedString("jetpack.fullscreen.overlay.notifications.continue.title",
                                                                             value: "Continue to Notifications",
                                                                             comment: "Title of a button that dismisses an overlay and displays the Notifications screen.")
+            static let continueButtonTitle = NSLocalizedString("jetpack.fullscreen.overlay.phaseThree.general.continue.title",
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel.swift:315]: [nit] 

```suggestion
            static let continueButtonTitle = NSLocalizedString("jetpack.fullscreen.overlay.general.continue.title",
```

**Review by momo-ozawa — COMMENTED** [review]
Works as described! A couple of UI things I noticed:

1. Are the buttons supposed to be leading aligned or center aligned in compact height mode? Currently they're aligned to the leading margin, but just wanted to make sure.

Simulator Screen Shot - iPhone 14 Pro - 2023-01-10 at 11 28 00

2. If you adjust the text size to use larger accessibility sizes, the labels in the secondary view overlap each other.

**Comment by hassaanelgarem** [comment]
> Are the buttons supposed to be leading aligned or center aligned in compact height mode? Currently they're aligned to the leading margin, but just wanted to make sure.

@momo-ozawa That's the expected behavior, yes. Below is a screenshot from Figma:

iOS Overlay

> If you adjust the text size to use larger accessibility sizes, the labels in the secondary view overlap each other.

Nice catch! This is fixed in 155a642cfa6c254c90f94096a12dcbafbe673aee & 483007d2bb27103f50f9ebb3b0f12843f439e4eb 👍

File: WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel.swift (hunk 1 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel.swift:315]: I intentionally did not update this to not force a new translation of an already translated string. But now I'm having second thoughts. Not sure what's worse, having an inaccuracy in the key or translating an already translated string. What do you think?
Related thread: p1650995535103199/1650995010.603439-slack-CC7L49W13

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Jetpack/Branding/Fullscreen Overlay/JetpackFullscreenOverlayGeneralViewModel.swift:315]: Thanks for the thread link, that makes a lot of sense.

Hmmm. In this case the renaming is a super minor change (i.e. it doesn't noticeably improve the readability etc), so maybe it's best to leave it as is.

**Comment by momo-ozawa** [comment]
> Nice catch! This is fixed in [155a642](https://github.com/wordpress-mobile/WordPress-iOS/commit/155a642cfa6c254c90f94096a12dcbafbe673aee) & [483007d](https://github.com/wordpress-mobile/WordPress-iOS/commit/483007d2bb27103f50f9ebb3b0f12843f439e4eb) 👍

Re-tested, and can confirm that overlapping issue is resolved! Thanks 🙇‍♀️

**Review by momo-ozawa — APPROVED** [review]

**Pull request merged**